libmscore/measure.cpp

```diff
@@ -3418,9 +3418,17 @@ qreal Measure::createEndBarLines(bool isLastMeasureInSystem)
             seg->setEnabled(false);
             }
       else {
-            BarLineType t = nm ? BarLineType::NORMAL : BarLineType::END;
+            BarLineType t = BarLineType::NORMAL;
             if (!seg)
                   seg = getSegmentR(SegmentType::EndBarLine, ticks());
+            else {
+                  //A segment exists, therefore we should continue this barline type
+                  BarLine* bl = toBarLine(seg->element(0));
```

Comment:
I guess a possible complaint is, why single out element(0)?  It *could* be the case that the top staff has a double bar the next does not, etc.  If we're to inherit an existing barline style, maybe better to pick on the staff just above?